### ISSUE TITLE
Cannot find `remark-parse` in remark-language-server

### ISSUE THREAD

**Issue by s417-lama** [mattn/vim-lsp-settings#527]
I tried remark-language-server added by in @remcohaszing in #516, but I got the following error.

```
Could not find module remark-parse
Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'remark-parse' imported from /path/to/workingdir
    at node_internal_ (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/errors.js:315:11)
    at __node_internal_ (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/errors.js:258:13)
    at new NodeError (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/errors.js:242:5)
    at packageResolve (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/resolve.js:968:9)
    at moduleResolve (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/resolve.js:1024:18)
    at defaultResolve (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/lib/resolve.js:1062:13)
    at resolve (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/import-meta-resolve/index.js:27:12)
    at attempt (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/load-plugin/index.js:179:15)
    at resolvePlugin (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/load-plugin/index.js:160:16)
    at addModule (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:338:18)
    at use (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:319:13)
    at addEach (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:288:53)
    at addPreset (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:263:39)
    at merge (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:245:11)
    at Configuration.create (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:192:11)
    at file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/configuration.js:137:14
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at apply (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:241:9)
    at applyAll (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:227:9)
    at found (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:215:9)
    at next (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:152:13)
    at ReadFileContext.done [as callback] (file://[$HOME]/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/unified-engine/lib/find-up.js:174:20)
    at FSReqCallback.readFileAfterOpen [as oncomplete] (fs.js:261:13)
```

Seems like `remark-parse` package is not installed under `~/.local/share/vim-lsp-settings/servers/remark-language-server/node_modules/`.

Although I'm not sure if it is an issue in vim-lsp-settings or in the original remark-language-server, I posted it here for now.

**Comment by remcohaszing** [comment]
The same issue was reported in https://github.com/neovim/nvim-lspconfig/pull/1606#discussion_r780224035 and it’s being worked on.

**Comment by rubin55** [comment]
@s417-lama can you install the `remark-parse` npm package into `~/.local/share/vim-lsp-settings/servers/remark-language-server` as a workaround? Does that work?

**Comment by nshen** [comment]
same here 

Counld not find `remark-parse`

**Comment by remcohaszing** [comment]
> @s417-lama can you install the `remark-parse` npm package into `~/.local/share/vim-lsp-settings/servers/remark-language-server` as a workaround? Does that work?

There’s no need to install `remark-parse` in the language server. It was trying to load it from your local `node_modules`.

`remark-language-server` has just been updated. It now loads `remark` instead.

Unfortunately it’s now crashing on my machine and I cant get logging to work in `vim-lsp`.

**Comment by s417-lama** [comment]
Unfortunately it did not work.

I executed:
```
~/.local/share/vim-lsp-settings/servers/remark-language-server
⟩ npm install remark-parse
```

and reopened vim, but no luck.

**Comment by remcohaszing** [comment]
> Unfortunately it’s now crashing on my machine and I cant get logging to work in `vim-lsp`.

My bad, this was caused by me using an older local version of `remark`

@s417-lama please update `remark-language-server` to the latest version. You do not need to install `remark-parse`

**Comment by nshen** [comment]
![image](https://user-images.githubusercontent.com/181506/149140126-dbe6f9c5-b586-4ee4-9206-168a49498a5d.png)

what should I do @remcohaszing ?

**Comment by s417-lama** [comment]
Same here. The original error was resolved by updating `remark-language-server`, but the above error was also shown in my environment.
`npm install remark` did not help.

**Comment by rubin55** [comment]
> ![image](https://user-images.githubusercontent.com/181506/149140126-dbe6f9c5-b586-4ee4-9206-168a49498a5d.png)

Mm, shouldn't that message say `npm -g install remark`?

**Comment by Urfoex** [comment]
Added packages:
```
"$(dirname "$0")/npm_install.sh" remark-language-server remark-language-server
"$(dirname "$0")/npm_install.sh" remark remark-cli                            
npm install remark                                                            
```
Still I'm getting 'Cannot turn on language server without `remark` locally. Run `npm install remark` to enable it'

```
vim-lsp-settings/servers/remark-language-server$
remark -> ./node_modules/.bin/remark
remark-language-server -> ./node_modules/.bin/remark-language-server
```

```
vim-lsp-settings/servers/node_modules/.bin$
remark -> ../remark-cli/cli.js
remark-language-server -> ../remark-language-server/index.js
```

```
vim-lsp-settings/servers/node_modules$
remark
remark-cli
remark-language-server
```

**Comment by rubin55** [comment]
Just tried, keep getting:

`Cannot turn on language server without `remark` locally. Run `npm install remark` to enable it`

It doesn't matter if I `npm install remark` within the `vim-lsp-settings/servers/remark-language-server` folder, or if I `npm install -g remark`. 

What obvious thing am I doing wrong?

**Comment by fasterius** [comment]
I am also getting this exact same error (`Cannot turn on language server without `remark` locally. Run `npm install remark` to enable it`), with no idea how to get it to work. Is there a solution to this yet?

**Comment by halostatue** [comment]
There is no solution to this, and I do not believe that there will be a solution from the `remark-language-server` side. See the notes that I added to #543  where I recommend that `remark-language-server` be further restricted to not be suggested except to those few people who have already opted into `remark` and removed if that cannot be achieved with `vim-lsp-settings`.

**Comment by fasterius** [comment]
Thanks for the update!

**Comment by halostatue** [comment]
This can probably be closed because of #545.
